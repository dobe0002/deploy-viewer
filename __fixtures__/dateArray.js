export default [
  new Date(2019, 10, 1), // Fri Nov 01 2019 00: 00: 00 GMT - 0500(Central Daylight Time),
  new Date(2019, 10, 2), // Sat Nov 02 2019 00:00:00 GMT-0500 (Central Daylight Time),
  new Date(2019, 10, 3), // Sun Nov 03 2019 00:00:00 GMT-0500 (Central Daylight Time),
  new Date(2019, 10, 4), // Mon Nov 04 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 5), // Tue Nov 05 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 6), // Wed Nov 06 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 7), // Thu Nov 07 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 8), // ri Nov 08 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 9), // Sat Nov 09 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 10), // Sun Nov 10 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 11), // Mon Nov 11 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 12), // Tue Nov 12 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 13), // Wed Nov 13 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 14), // Thu Nov 14 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 15), // Fri Nov 15 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 16), // Sat Nov 16 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 17), // Sun Nov 17 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 18), // Mon Nov 18 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 19), // Tue Nov 19 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 20), // Wed Nov 20 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 21), // Thu Nov 21 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 22), // Fri Nov 22 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 23), // Sat Nov 23 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 24), // Sun Nov 24 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 25), // Mon Nov 25 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 26), // Tue Nov 26 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 27), // Wed Nov 27 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 28), // Thu Nov 28 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 29), // Fri Nov 29 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 10, 30), // Sat Nov 30 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 11, 1), // Sun Dec 01 2019 00:00:00 GMT-0600 (Central Standard Time),
  new Date(2019, 11, 2) // Mon Dec 02 2019 00:00:00 GMT-0600 (Central Standard Time)
];
