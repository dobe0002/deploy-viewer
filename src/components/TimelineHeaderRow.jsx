import React from 'react';
import PropTypes from 'prop-types';
import { getMonth, getDate, format } from 'date-fns';

export default class TimelineHeaderRow extends React.Component {
  monthRow() {
    const { dateArray } = this.props;
    const html = [];

    let currentMonth = getMonth(dateArray[0]);

    let currentCount = 0;

    dateArray.forEach(date => {
      if (getMonth(date) !== currentMonth) {
        html.push(
          <td colSpan={currentCount}>
            {format(new Date().setMonth(currentMonth), 'MMMM')}
          </td>
        );
        currentMonth = getMonth(date);
        currentCount = 1;
      } else currentCount += 1;
    });
    // need to capture the last:
    html.push(
      <td colSpan={currentCount}>
        {format(new Date().setMonth(currentMonth), 'MMMM')}
      </td>
    );

    return html;
  }

  render() {
    const { dateArray } = this.props;
    return (
      <thead>
        <tr>
          <td />
          {this.monthRow()}
        </tr>
        <tr>
          <td />
          {dateArray.map((date, i) => (
            // eslint-disable-next-line react/no-array-index-key
            <th key={i}>{getDate(date)}</th>
          ))}
        </tr>
      </thead>
    );
  }
}

TimelineHeaderRow.propTypes = {
  dateArray: PropTypes.string
};
TimelineHeaderRow.defaultProps = {
  dateArray: []
};
