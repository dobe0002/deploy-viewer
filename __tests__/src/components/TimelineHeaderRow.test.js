/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TimelineHeaderRow from '../../../src/components/TimelineHeaderRow';
import dateArray from '../../../__fixtures__/dateArray';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Timeline Header Row component tests', () => {
  const sampleProps = {
    dateArray
  };
  test('Accessibility check', async () => {
    const wrapper = shallow(
      <TimelineHeaderRow dateArray={sampleProps.dateArray} />
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
  test('Snapshot test', () => {
    const wrapper = shallow(
      <TimelineHeaderRow dateArray={sampleProps.dateArray} />
    );

    // There are Two months (and a blank cell)
    expect(
      wrapper
        .find('tr')
        .first()
        .find('td')
    ).toHaveLength(3);

    // number of dates
    expect(
      wrapper
        .find('tr')
        .at(1)
        .find('th')
    ).toHaveLength(32);
    // expect(wrapper).toMatchSnapshot();
  });
});
