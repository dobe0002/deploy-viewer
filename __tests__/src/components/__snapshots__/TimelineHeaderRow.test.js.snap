// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline component tests Snapshot test 1`] = `
<thead>
  <tr>
    <td />
    <td
      colSpan={30}
    >
      November
    </td>
    <td
      colSpan={2}
    >
      December
    </td>
  </tr>
  <tr>
    <td />
    <th
      key="0"
    >
      1
    </th>
    <th
      key="1"
    >
      2
    </th>
    <th
      key="2"
    >
      3
    </th>
    <th
      key="3"
    >
      4
    </th>
    <th
      key="4"
    >
      5
    </th>
    <th
      key="5"
    >
      6
    </th>
    <th
      key="6"
    >
      7
    </th>
    <th
      key="7"
    >
      8
    </th>
    <th
      key="8"
    >
      9
    </th>
    <th
      key="9"
    >
      10
    </th>
    <th
      key="10"
    >
      11
    </th>
    <th
      key="11"
    >
      12
    </th>
    <th
      key="12"
    >
      13
    </th>
    <th
      key="13"
    >
      14
    </th>
    <th
      key="14"
    >
      15
    </th>
    <th
      key="15"
    >
      16
    </th>
    <th
      key="16"
    >
      17
    </th>
    <th
      key="17"
    >
      18
    </th>
    <th
      key="18"
    >
      19
    </th>
    <th
      key="19"
    >
      20
    </th>
    <th
      key="20"
    >
      21
    </th>
    <th
      key="21"
    >
      22
    </th>
    <th
      key="22"
    >
      23
    </th>
    <th
      key="23"
    >
      24
    </th>
    <th
      key="24"
    >
      25
    </th>
    <th
      key="25"
    >
      26
    </th>
    <th
      key="26"
    >
      27
    </th>
    <th
      key="27"
    >
      28
    </th>
    <th
      key="28"
    >
      29
    </th>
    <th
      key="29"
    >
      30
    </th>
    <th
      key="30"
    >
      1
    </th>
    <th
      key="31"
    >
      2
    </th>
  </tr>
</thead>
`;
